File: SauceDemo/PageObjects/HomePage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading;

namespace SeleniumAutomation.PageObjects
{
    public class HomePage : BasePage
    {
        public HomePage(IWebDriver driver) : base(driver)
        {
        }

        private IWebElement TB_name => HelperMethods.FindElementWithWait_Clickable(_driver, By.Name("username"), TimeSpan.FromSeconds(10));
        private IWebElement TB_pw => HelperMethods.FindElementWithWait_Clickable(_driver, By.Name("password"), TimeSpan.FromSeconds(10));
        private IWebElement BUT_bill_pay => HelperMethods.FindElementWithWait_Clickable(_driver, By.XPath("//li[a[text()='Bill Pay']]"), TimeSpan.FromSeconds(10));
        private IWebElement BUT_register => HelperMethods.FindElementWithWait_Clickable(_driver, By.LinkText("Register"), TimeSpan.FromSeconds(10));

        public void gotoHomePage()
        {
            Goto("");
            Thread.Sleep(3000);
        }

        public void Login(string name, string pw)
        {
            TB_name.SendKeys(name);
            TB_pw.SendKeys(pw);
            TB_pw.Submit();
        }

        public void BypassLoginWithCookie()
        {
            Cookie ck = new Cookie("session-username", "standard_user");
            _driver.Manage().Cookies.AddCookie(ck);
        }
        public void GoToRegistrationPage()
        {
            BUT_register.Click();
        }
    }
}

File: Selenium_Python/pages/homepage.py
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.chrome.options import Options
#from selenium.webdriver.common.cookie import Cookie
from helper import helper
from pages.basepage import basepage

import time

class homepage(basepage):
    def __init__(self, driver):
        super().__init__(driver)

    @property
    def TB_name(self):
        return helper.find_element_with_wait_clickable(self._driver, By.NAME, "username", 10)

    @property
    def TB_pw(self):
        return helper.find_element_with_wait_clickable(self._driver, By.NAME, "password", 10)

    def goto_home_page(self):
        self.goto("")
        time.sleep(3)

    def login(self, name, pw):
        self.TB_name.send_keys(name)
        self.TB_pw.send_keys(pw)
        self.TB_pw.send_keys(Keys.RETURN)

    def bypass_login_with_cookie(self):
        cookie = {'name': 'session-username', 'value': 'standard_user'}
        self._driver.add_cookie(cookie)

